CHANGELOG.md

```diff
@@ -1,3 +1,112 @@
+  import the `MAP` module now, instead of only the main module.
+
+- The new Haskell backend hooks for `encodeBytes` and `decodeBytes` are
+  exposed in `domains.md`.
+
+- Added support for module signatures supporting public/private syntax and imports.
+
+- `macro` is now expected to be added to the syntax declaration instead of rules.
+
+- Rework timing info for `--profile-rule-parsing`. It now takes a file and
+  includes separate data fields for parsing and type inference.
+
+- Added `kprovex`. A lightweight version of `kprove` which restricts the spec files
+  to claims and token syntax. This avoids several bugs in the old code path and
+  slightly improves the front end overhead. To convert the old tests, you need to
+  kompile ahead of time the definition with the simplification rules and extra
+  helper syntax needed for the proofs.
+
+Documentation
+-------------
+
+- Relicence as BSD 3-Clause
+
+- Lessons Basic 2-21 of the new K tutorial have been written.
+
+- The option `--help-experimental` or `-X` is removed from all tools, and 
+  all the `--help` option lists are flattened.
+
+- Rename `pending-documentation.md` into `USER_MANUAL.md`
+
+Misc/Bug Fixes
+--------------
+
+- Modules marked as `not-lr1` are automatically excluded from the generated
+    Bison parser, instead of causing an error.
+
+- `krun` respects the `--no-exc-wrap` option now.
+
+- Formatting of several error messages is improved, especially indentation.
+
+- File symlink handling in `requires` statements is improved.
+
+- `kast` tool no longer silently ignores the `--output-file` parameter.
+
+- Verbose mode offers more information about timings and scanners being created.
+
+- Added `-I`, `--md-selector` and `--no-prelude` to kprove, kprovex and kbmc
+
+Performance Improvements
+------------------------
+
+- Definition parsing performance is significantly improved by being smarter
+  with the existing caches and optimizing some specific code hotspots.
+
+- Improve type inference time for anonymous variables.
+
+- Parse Record Productions in linear time.
+
+- The K formula calculating Array extension has been optimized in `domains.md`.
+
+Packaging/Deployment
+--------------------
+
+-   The Nix packages are built with LTO turned on. Bison is included
+    in the Nix package dependencies. `kserver` now works on Nix.
+
+-   The platform independent K binary, which really would only realistically
+    work with Ubuntu Bionic (18.04), has been removed as an install option.
+
+-   The Arch linux Dockerfiles have been updated for new Arch docker image
+    naming scheme.
+
+
+Dependency Updates
+------------------
+
+-   Haskell backend is updated to version d4239d97f5b53188460e7399b683111b5687661c.
+
+-   LLVM backend is updated to version 23c2adcd21a1486442062ecc4ae051548215bcc3.
+
+-   K Web Theme is updated to version b458d1461c31760024ea06e0e50f25806ace5e2c.
+
+-   JavaCC is updated to version 4.1.4.
+
+-   commons-io Java library is bumped to version 2.7.
```

Comment:
The indentation here on list entries is different to above - is this deliberate?